lib/runner/cli/cli.js

```diff
@@ -331,8 +332,14 @@ class CliRunner {
     return this;
   }
 
-  getTestsFiles() {
-    return Runner.readTestSource(this.test_settings, this.argv);
+  async getTestsFiles() {
+    const modules = {};
+    for (const env of this.testEnvArray) {
+      const envSettings = Settings.parse(this.userSettings, this.baseSettings, this.argv, env);
+      modules[env] = await Runner.readTestSource(envSettings, this.argv);
+    }
+
```

Comment:
What's the benefit of having modules divided as per env. As for all the modules found we need to run it against every env given.